tools/testsys/src/run.rs

```diff
@@ -255,6 +258,48 @@ impl Run {
                     .build()
                     .context(error::DatacenterBuildSnafu)?;
 
+                let vsphere_secret = if !variant_config.secrets.contains_key("vsphereCredentials") {
+                    info!("Creating vSphere secret, 'vsphereCreds'");
+                    let creds_env = DatacenterCredsBuilder::from_env();
+                    let creds_file = if let Some(ref creds_file) = *VMWARE_CREDS_PATH {
+                        if creds_file.exists() {
+                            info!("Using vSphere credentials file at {}", creds_file.display());
+                            DatacenterCredsConfig::from_path(creds_file)
+                                .context(error::VmwareConfigSnafu)?
+                        } else {
+                            info!(
+                            "vSphere credentials file not found, will attempt to use environment"
+                        );
+                            DatacenterCredsConfig::default()
+                        }
+                    } else {
+                        info!("Unable to determine vSphere credentials file location, will attempt to use environment");
+                        DatacenterCredsConfig::default()
+                    };
+                    let dc_creds = creds_file.datacenter.get(&datacenter.datacenter);
+                    let creds: DatacenterCreds = creds_env
+                        .take_missing_from(dc_creds)
+                        .build()
+                        .context(error::CredsBuildSnafu)?;
+
+                    let secret_name =
+                        SecretName::new("vsphereCreds").context(error::SecretNameSnafu {
+                            secret_name: "vsphereCreds",
+                        })?;
+                    client
+                        .create_secret(
+                            &secret_name,
+                            vec![
+                                ("username".to_string(), creds.username),
+                                ("password".to_string(), creds.password),
+                            ],
+                        )
+                        .await?;
```

Comment:
What happens if the secret already exists? Can we make sure it replaces it if it already exists? (upsert)

Reply:
It should be ok. I will verify it when testing.